static-assets/components/cstudio-common/common-api.js

```diff
@@ -919,15 +919,79 @@ var nodeOpen = false,
 
             },
 
-            submitContent: function(site, contentItems) {
+            submitContent: function(site, items) {
 
                 CSA.Operations._showDialogueView({
                     fn: CSA.Service.getRequestPublishView,
                     controller: 'viewcontroller-requestpublish',
-                    callback: function(dialog) {
+                    callback: function(dialogue) {
+                        var self = this;
                         CSA.Operations.translateContent(formsLangBundle, ".cstudio-dialogue");
-                        this.renderItems(contentItems);
+                        this.loadItems(items);
                         this.loadPublishingChannels();
+
+                        CStudioAuthoring.Utils.removeLoadingIcon();
+
+                        this.on("submitComplete", function(evt, args){
```

Comment:
- Move to dialog util function
- Clean the callback to move contained logic to the dialog

Reply:
Done